core/runner.py

```diff
@@ -29,13 +29,11 @@ class DMLRunner(object):
 
     """
 
-    def __init__(self):
+    def __init__(self, config_manager):
```

Comment:
So I see that now we're passing the config manager instance and calling `reset`. I like the change, since it makes the tests a lot cleaner at the expense of little for the actual classes. 

But I wanted to ask (nitpick): if we're going to go as far as to pass the instance around, can we just make the config manager not a singleton? It feels like we're passing around a config file where each class uses a different part of the config. Why don't we just have separate configs for separate classes? For example, `DMLRunner` could have `configuration_runner.ini`, and so forth. Then each class creates its own instance of the config manager with respect to its own config file. I feel like testing in the future would significantly easier, depending on how many more components of the Unix Service require their own set of configurations.

Reply:
I agree with you. If we're doing Dependency Injection we should probably not do Singleton too, as it opens up 2 ways of accessing the class: by passing it around and by calling the get instance static method.

The benefit of DI over Singleton is that testing is a lot easier and probably readability too, so I say let's stick to DI, make the class a non-Singleton, and make the fact that there is only 1 Configuration Manager an implementation detail.

In regards to having separate configurations, I don't really see a benefit there. The idea of having only 1 configuration with multiple sections adds more ease of use and is less complex in my opinion.

Now, this makes me reconsider the design choice of making Dataset Manager a Singleton. Maybe we should change it and make the (future) Listener inject the DM...

Reply:
Okay, the class is not a singleton anymore. Since there's still only 1 Configuration Manager, I still think it makes sense to use `get_instance`. So tl;dr just removed the enforcement of only one instance.

Totally agree that DI is easier. Especially this case, since we're using the Configuration Manager in basically three different ways. 

Inject the DM in the Scheduler -> Runners, yes? That makes sense, but we can make an official decision once we work on the Listener.

If everything looks good, I can go ahead and merge?

Reply:
Let's go ahead and remove the `get_instance()` and `_instance` property so we don't get confused thinking the class is a Singleton in the future. This also comes with changing the tests to initialize the class normally. After that we're ready to merge.

Reply:
Sorry about the delay. Made the changes you asked for and I will be merging now.